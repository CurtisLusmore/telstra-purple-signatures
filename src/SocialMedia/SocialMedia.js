import React, { Component } from 'react';

class SocialMedia extends Component {
  static brandInfo = [
    {
      name: 'Facebook',
      link: 'https://readify.net/links/facebook',
      image:
        'https://readifysignatures.blob.core.windows.net/images/image002.png',
      alt: 'Readify on Facebook'
    },
    {
      name: 'Twitter',
      link: 'https://readify.net/links/twitter',
      image:
        'https://readifysignatures.blob.core.windows.net/images/image003.png',
      alt: 'Readify on Twitter'
    },
    {
      name: 'LinkedIn',
      link: 'https://readify.net/links/linkedin',
      image:
        'https://readifysignatures.blob.core.windows.net/images/image004.png',
      alt: 'Readify on LinkedIn'
    },
    {
      name: 'YouTube',
      link: 'https://readify.net/links/youtube',
      image:
        'https://readifysignatures.blob.core.windows.net/images/image005.png',
      alt: 'Readify on YouTube'
    }
  ];

  static htmlLinks = () =>
    SocialMedia.brandInfo.map((mediaObj, index, arr) => (
      <span key={mediaObj.name}>
        <a
          style={{ display: 'inline-block', width: '28px', height: '28px' }}
          href={mediaObj.link}
        >
          <img width="28" height="28" src={mediaObj.image} alt={mediaObj.alt} />
        </a>
        {index < arr.length - 1 ? <span>&nbsp;&nbsp;</span> : null}
      </span>
    ));

  render() {
    return <p style={{ marginBottom: '0' }}>{SocialMedia.htmlLinks()}</p>;
  }
}

export default SocialMedia;
