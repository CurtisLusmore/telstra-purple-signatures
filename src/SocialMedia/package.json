{
  "main": "./SocialMedia.js",
  "name": "SocialMedia",
  "version": "0.0.0",
  "private": true
}