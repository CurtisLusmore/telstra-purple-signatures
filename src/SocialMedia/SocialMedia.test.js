import React from 'react';
import { shallow } from 'enzyme';

import SocialMedia from './SocialMedia';

it('Renders', () => {
  const tree = shallow(<SocialMedia />);
  expect(tree).toMatchSnapshot();
});
