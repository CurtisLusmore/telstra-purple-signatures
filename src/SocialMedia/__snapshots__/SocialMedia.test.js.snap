// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<p
  style={
    Object {
      "marginBottom": "0",
    }
  }
>
  <span
    key="FaceBook"
  >
    <a
      href="https://readify.net/links/facebook"
      style={
        Object {
          "display": "inline-block",
          "height": "28px",
          "width": "28px",
        }
      }
    >
      <img
        alt="Readify on FaceBook"
        height="28"
        src="https://readifysignatures.blob.core.windows.net/images/image002.png"
        width="28"
      />
    </a>
    <span>
        
    </span>
  </span>
  <span
    key="Twitter"
  >
    <a
      href="https://readify.net/links/twitter"
      style={
        Object {
          "display": "inline-block",
          "height": "28px",
          "width": "28px",
        }
      }
    >
      <img
        alt="Readify on Twitter"
        height="28"
        src="https://readifysignatures.blob.core.windows.net/images/image003.png"
        width="28"
      />
    </a>
    <span>
        
    </span>
  </span>
  <span
    key="LinkedIn"
  >
    <a
      href="https://readify.net/links/linkedin"
      style={
        Object {
          "display": "inline-block",
          "height": "28px",
          "width": "28px",
        }
      }
    >
      <img
        alt="Readify on LinkedIn"
        height="28"
        src="https://readifysignatures.blob.core.windows.net/images/image004.png"
        width="28"
      />
    </a>
    <span>
        
    </span>
  </span>
  <span
    key="Youtube"
  >
    <a
      href="https://readify.net/links/youtube"
      style={
        Object {
          "display": "inline-block",
          "height": "28px",
          "width": "28px",
        }
      }
    >
      <img
        alt="Readify on Youtube"
        height="28"
        src="https://readifysignatures.blob.core.windows.net/images/image005.png"
        width="28"
      />
    </a>
  </span>
</p>
`;
